src/Data/Types/Enumeration.php

```diff
@@ -0,0 +1,65 @@
+<?php
+namespace LORIS\Data\Types;
+
+/**
+ * A Enumeration represents a data type where the value must be one of
+ * a specific list of option.
+ *
+ * @license http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ */
+class Enumeration implements \LORIS\Data\Type
+{
+    protected $options = [];
+
+    /**
+     * Construct an Enumeration type
+     *
+     * @param string ...$values The options for this enumeration
+     */
+    public function __construct(string ...$values)
```

Comment:
out of curiosity, why the splat operator instead of an array ?

Reply:
I find in this case it just looks more aesthetically pleasing to do ie `new Enumeration('Male', 'Female', 'Other')`.

On a more technical level it also ensures that it's an array of strings, while `array` would be any arbitrary array with no type enforcement (for the key/values)

Reply:
I guess I have 3 concerns:
1. not sure what code is currently using it but in the event where we try to build the enum on the go, an array offers the advantage of being able to append and prepend until we are ready to create the enum. if each element needs to be specified manually it makes creating the enum progressively almost impossible.
2. it seems much more difficult to build the enum from the results of a query, especially if the enum is of a large list
3. why do we only want strings? are int enums not an option ?

Reply:
For 1 and 2: I don't see why that's a problem. You can build an array as normal and then use the splat operator to expand the array such as here: https://github.com/aces/Loris/pull/6936/files#diff-88088f05f1a41366915499faede2a27ee316859e38c7990a86662680c1b89f4eR289

For 3: no. At least not from a MySQL perspective.

```
mysql> create temporary table foo ( id enum(2, 3) );
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '2, 3) )' at line 1
mysql> create temporary table foo ( id enum('2', '3') );
Query OK, 0 rows affected (0.01 sec)
```

from a conceptual perspective enums are neither ints nor strings. They're "one of a set of things" independent of the internal representation, but we always want the labels used to identify them to be strings (much like mysql does in the above example..)

Reply:
Yeah for 1 and 2 I edited my comment. I only realised it after.

3 makes sens as well, i guess I was thinking more of PHP arrays than SQL enums.